llama-index-core/llama_index/core/readers/base.py

```diff
@@ -71,6 +71,141 @@ class Config:
         arbitrary_types_allowed = True
 
 
+class BaseResourcesReader(BaseReader):
```

Comment:
This looks good to me, and probably fine to merge for now.

But in general, nesting base class under base class can be a confusing pattern to maintain 😅 Longer term I'd prefer just a single base class, or mixin's that add functionality 

Because now we have BaseReader -> BaseResourcesReader -> BaseFileSystemReader -> <Actual reader implemention>